clinica/pydra/statistics_volume_correction/pipeline.py

```diff
@@ -0,0 +1,181 @@
+import pydra
+from nipype.algorithms.misc import Gunzip
+from pydra import Workflow
+from pydra.tasks.nipype1.utils import Nipype1Task
+
+from clinica.pydra.engine import clinica_io
+
+
+@clinica_io
+def build_core_workflow(name: str = "core", parameters={}) -> Workflow:
+    """Build the core workflow for the Statistics Volume pipeline.
+
+    Parameters
+    ----------
+    name : str, optional
+        The name of the workflow. Default="core".
+
+    parameters : dict, optional
+        Dictionary of parameters to be used
+        within the workflow.
+        Default={}.
+
+    Returns
+    -------
+    wf : Workflow
+        The core workflow.
+    """
+    from os.path import abspath, dirname, exists, join, pardir
+    from typing import Any
+
+    import numpy as np
+
+    import clinica.pydra.statistics_volume_correction.task as utils
+    from clinica.utils.exceptions import ClinicaException
+    from clinica.utils.inputs import RemoteFileStructure, fetch_file
+    from clinica.utils.spm import spm_standalone_is_available, use_spm_standalone
+    from clinica.utils.stream import cprint
+
+    if spm_standalone_is_available():
+        use_spm_standalone()
+
+    # parameters = _check_pipeline_parameters(parameters)
+
+    input_name = "t_map"
+    query = {"pattern": parameters["t_map"] + "*", "description": "statistics t map"}
+
+    input_spec = pydra.specs.SpecInfo(
+        name="Input",
+        fields=[
+            ("_graph_checksums", Any),
+            (input_name, dict, query, {"mandatory": True}),
+        ],
+        bases=(pydra.specs.BaseSpec,),
+    )
+    wf = Workflow(name, input_spec=input_spec)
+    wf.add(
+        utils.peak_correction_task(
+            name="FWE_peak_correction_task",
+            t_map=wf.lzin.t_map,
+            t_threshold=parameters["FWEp"],
+        )
+    )
+    wf.add(
+        utils.peak_correction_task(
+            name="FDR_peak_correction_task",
+            t_map=wf.lzin.t_map,
+            t_threshold=parameters["FDRp"],
+        )
+    )
+    wf.add(
+        utils.cluster_correction_task(
+            name="FWE_cluster_correction_task",
+            t_map=wf.lzin.t_map,
+            t_thresh=parameters["height_threshold"],
+            c_thresh=parameters["FWEc"],
+        )
+    )
+    wf.add(
+        utils.cluster_correction_task(
+            name="FDR_cluster_correction_task",
+            t_map=wf.lzin.t_map,
+            t_thresh=parameters["height_threshold"],
+            c_thresh=parameters["FDRc"],
+        )
+    )
+    root = dirname(abspath(join(abspath(__file__), pardir, pardir)))
+    path_to_mask = join(root, "resources", "masks")
+    url_aramis = "https://aramislab.paris.inria.fr/files/data/img_t1_linear/"
+    FILE1 = RemoteFileStructure(
+        filename="mni_icbm152_t1_tal_nlin_sym_09a.nii.gz",
+        url=url_aramis,
+        checksum="3b244ee7e287319d36a25263744c468ef0ab2fe5a94b15a2138844db73b49adf",
```

Comment:
Any chance this is what you're looking for, already implemented and ready to be plugged ?

https://github.com/aramis-lab/clinica/blob/a499f53eea49fd4151db30642239c34d7d3fa323/clinica/pydra/tasks.py#L20-L28

Reply:
I'll check it out

Reply:
I'm not sure it is what I want. I implemented this part the way it was made before, without any task. It does work that way. 
It feels like this is part of the logic I want. Maybe some hints about how to work this out are in `t1-linear` ? I'll check it out

Reply:
The file returned by this function is not the right one. Maybe we can discuss this tomorrow ?

Reply:
You're right, it isn't the same file... 😓 The difference is minimal though (one letter actually...).

I think we should be consistent, so I would still implement this as a task function, either by passing a parameter to select the file, or by having two functions. Or something in between like this:

```python
@task
@annotate({"return": {"mni_template_file": PurePath}})
def download_mni_template_2009a() -> PurePath:
    return _download_mni_template(version="09a")

@task
@annotate({"return": {"mni_template_file": PurePath}})
def download_mni_template_2009c() -> PurePath:
    return _download_mni_template(version="09c")

def _download_mni_template(version: str) -> PurePath:
    return download_file(
        url=f"https://aramislab.paris.inria.fr/files/data/img_t1_linear/mni_icbm152_t1_tal_nlin_sym_{version}.nii.gz",
        to=str(Path.cwd() / f"mni_icbm152_t1_tal_nlin_sym_{version}.nii.gz"),
    )
```